src/task.c

```diff
@@ -395,6 +325,154 @@ task_thread_create(const char            *name,
    return STATUS_OK;
 }
 
+/******************************************************************************
+ * Registering/deregistering threads that were not started by the task
+ * system.
+ ******************************************************************************/
+
+/*
+ * task_register_thread(): Register this new thread with the task system.
+ *
+ * This is for use by threads created outside of the splinter system.
+ *
+ * Registration implies:
+ *  - Acquire a new thread ID (index) for this to-be-active thread
+ *  - Allocate scratch space for this thread, and track this space..
+ */
+platform_status
+task_register_thread(task_system *ts,
+                     uint64       scratch_size,
+                     const char  *file,
+                     const int    lineno,
+                     const char  *func)
+{
+   threadid thread_tid;
+
+   debug_code(thread_tid = platform_get_tid());
+   debug_assert(
+      thread_tid == INVALID_TID,
+      "[%s:%d::%s()] Attempt to register already registered thread %lu\n",
+      file,
+      lineno,
+      func,
+      thread_tid);
+
+   thread_tid = allocate_threadid(ts);
+   if (thread_tid == INVALID_TID) {
+      return STATUS_NO_SPACE;
+   }
+   platform_set_tid(thread_tid);
+
+   debug_assert(ts->thread_scratch[thread_tid] == NULL,
+                "Scratch space should not yet exist.");
+
+   if (0 < scratch_size) {
+      char *scratch = TYPED_MANUAL_ZALLOC(ts->heap_id, scratch, scratch_size);
+      if (scratch == NULL) {
+         platform_set_tid(INVALID_TID);
+         deallocate_threadid(ts, thread_tid);
+         return STATUS_NO_MEMORY;
+      }
+      ts->thread_scratch[thread_tid] = scratch;
+   }
+
+   task_run_thread_hooks(ts);
+
+   return STATUS_OK;
+}
+
+/*
+ * task_deregister_thread() - Deregister an active thread from the task system.
+ *
+ * Deregistration involves:
+ *  - Releasing any scratch space acquired for this thread.
+ *  - Clearing the thread ID (index) for this thread
+ */
+void
+task_deregister_thread(task_system *ts,
+                       const char  *file,
+                       const int    lineno,
+                       const char  *func)
+{
+   threadid tid = platform_get_tid();
+
+   debug_assert(tid != INVALID_TID,
```

Comment:
Can this be changed to `platform_assert()`? I think, as requested for the checks in registration, these events are not that frequent, so it's better to tighten these checks for release builds, too.